db/kv_dense_block.go

```diff
@@ -294,6 +330,39 @@ func (d *DenseBlock) removeEntries(entries []*LogEntry) []*LogEntry {
 
 }
 
+// Rollback removes any entries in the block for the vbucket greater than the specified sequence
+func (d *DenseBlock) rollbackEntries(vbNo uint16, seq uint64) (numRemoved int, rollbackComplete bool) {
+
+	// Work backwards through the block, removing entries greater than vbNo, seq
+	indexPos := 2 + uint32(d.getEntryCount()-1)*INDEX_ENTRY_LEN
```

Comment:
It's the block header (entry count).  I've been lazy about replacing all the '2' references with a constant - I've made that change now.